### PR TITLE
Minor cleanup of the translation rules for cumred primitives

### PR DESCRIPTION
**Pull request by gnecula**

### CODE REVIEW

File: jax/_src/lax/control_flow.py [hunk 1]
```diff
@@ -2574,39 +2574,27 @@ def _cumred_dtype_rule(name, operand, *args, **kw):
                     "of number.".format(name, np.dtype(operand.dtype).name))
   return dtypes.canonicalize_dtype(operand.dtype)
 
-cumsum_p = lax.standard_primitive(
```
Comment (gnecula) [inline jax/_src/lax/control_flow.py:2577]: This definition was confusing when I was trying to understand the CPU translation rules. I did not realize that the translation rule was overriden below in the file.

**Comment by gnecula** [comment]
@marcvanzee

File: jax/_src/lax/control_flow.py [hunk 2]
```diff
@@ -2574,39 +2574,27 @@ def _cumred_dtype_rule(name, operand, *args, **kw):
...
-ad.deflinear2(cumsum_p, _cumsum_transpose_rule)
-xla.backend_specific_translations['tpu'][cumsum_p] = xla.lower_fun(
-  partial(_cumred_tpu_translation_rule, lax._reduce_window_sum),
-  multiple_results=False)
-batching.primitive_batchers[cumsum_p] = partial(_cumred_batch_rule, cumsum_p)
-
-
-def _cumulative_reduction_primitive(name, reduce_window_fn):
+def _cumulative_reduction_primitive(name,
+                                    associative_reduce_window_fn,
```
Comment (hawkinsp) [inline jax/_src/lax/control_flow.py:2578]: Just `reduce_fn`? There's no `window` about it if we are using `associative_scan`.

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/lax/control_flow.py (hunk 2 above)
Comment (gnecula) [reply jax/_src/lax/control_flow.py]: Done

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**